src/services/charges.ts

```
@@ -0,0 +1,65 @@
+import { PEANUT_API_URL } from '@/constants'
+import { CreateChargeRequest, PaymentCreationResponse, TCharge, TRequestChargeResponse } from './services.types'
+
+export const chargesApi = {
+    create: async (data: CreateChargeRequest): Promise<TCharge> => {
+        const response = await fetch(`/api/proxy/charges`, {
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/json',
+            },
+            body: JSON.stringify(data),
+        })
+
+        if (!response.ok) {
+            throw new Error(`Failed to create charge: ${response.statusText}`)
+        }
+
+        return response.json()
+    },
```

Q: Do we need this code for every single call? Can't we DRY a bit here?

Also I'd rather just raise the error message instead of adding a useless "Failed to create charge" infront

---

wdym, i did not get this

> Do we need this code for every single call?

do you mean, that content type, body, method type specification to be reused somehow ?

---

yes, i meant that. it's a very nitpick issue though and maybe bad approach